api/v1alpha1/storageconsumer_types.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2021 Red Hat OpenShift Container Storage.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StorageConsumerState represent a StorageConsumer's state
+type StorageConsumerState string
+
+const (
+	// StateReady represents Ready state of an object
+	StateReady StorageConsumerState = "Ready"
+	// StateConfiguring represents Configuring state of an object
+	StateConfiguring StorageConsumerState = "Configuring"
+	// StateDeleting represents Deleting state of an object
+	StateDeleting StorageConsumerState = "Deleting"
+	// StateFailed represents Failed state of an object
+	StateFailed StorageConsumerState = "Failed"
```

Comment:
Change all instances of `State<Type>` to `StorageConsumerState<Type>`. This can get pretty long, yes, but we don't yet have a standardized convention of "State" names across the project, so we need to be specific for now.